### PR TITLE
Feat/3.8 allow to edit publications

### PR DESCRIPTION
**Pull request by asf-algoo**
This pr allow a user to edit its own publications or a space manager to edit any publication.

## Checkpoints

**For developers**

- [x] If relevant, manual tests have been done to ensure the stability of the whole application and that the involved feature works
~~- [ ] The original issue is up to date w.r.t the latest discussions and contains a short summary of the implemented solution~~
It does not have an original issue. Integration feature.
- [ ] Automated tests covering the feature or the fix, have been written, deemed irrelevant (give the reason), or an issue has been created to implement the test (give the link)
- [x] Make sure that:
  - if there are modifications in the Tracim configuration files (eg. `development.ini`), they are documented in `backend/doc/setting.md`
  - any migration process required for existing instances is documented
  - relevant people for these changes are notified
- [x] Original authors of the features included in a multi-feature branch (maintenance fixes -> develop, security fixes -> develop, …) should be part of the reviewers, especially if you encountered merge conflicts.

**For code reviewers**

- [ ] The code is clear enough
- [ ] If there are FIXMEs in the code, related issues are mentioned in the FIXME
- [ ] If there are TODOs, NOTEs or HACKs in code, the date and the developer initials are present

**For testers**

~~- [ ] Manual, quality tests have been done~~
Manual tests not required during integration phase.

### CODE REVIEW

File: frontend/src/reducer/publicationList.js [hunk 1]
```diff
@@ -35,14 +36,18 @@ export default function publicationList (state = defaultPublicationList, action)
     case `${UPDATE}/${WORKSPACE_PUBLICATION_LIST}`:
       return uniqByContentId(sortByModifiedDate(state))
 
-    case `${UPDATE}/${PUBLICATION}`:
-      return uniqByContentId(state.map(publication => action.publication.content_id === publication.id
+    case `${UPDATE}/${PUBLICATION}`: {
+      const serializedPublication = action.publication.content_id
+        ? serialize(action.publication, serializeContentProps)
+        : action.publication
```
Comment (Skylsmoi) [inline frontend/src/reducer/publicationList.js:42]: what's the purpose of this test?

File: frontend_lib/src/component/Timeline/Timeline.jsx [hunk 2]
```diff
@@ -64,13 +64,12 @@ export class Timeline extends React.Component {
     this.setState({ showEditCommentPopup: true, newComment: comment })
   }
 
-  handleClickValidateEditComment = async () => {
-    const { props, state } = this
-
+  handleClickValidateEditComment = async (comment) => {
```
Comment (Skylsmoi) [inline frontend_lib/src/component/Timeline/Timeline.jsx:67]: shouldn't this change be reflected on every apps?

**Review by raphj — APPROVED** [review]

File: frontend/src/reducer/publicationList.js (hunk 1 above)
Comment (gbonaspetti) [reply frontend/src/reducer/publicationList.js:42]: test if the publication is serialized or not

File: frontend_lib/src/component/Timeline/Timeline.jsx (hunk 2 above)
Comment (gbonaspetti) [reply frontend_lib/src/component/Timeline/Timeline.jsx:67]: yes, that's why I changed on Timeline component

Comment (Skylsmoi) [reply frontend_lib/src/component/Timeline/Timeline.jsx:67]: indeed

**Review by Skylsmoi — APPROVED** [review]

**Pull request merged**